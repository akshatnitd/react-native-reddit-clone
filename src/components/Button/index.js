import React from "react"
import PropTypes from "prop-types"
import {
  Text,
  View,
  TouchableNativeFeedback,
  TouchableHighlight,
  Platform,
  StyleSheet,
} from "react-native"
import { colorStyle, fontStyle } from "../../style"

const Button = (props) => {
  const {
    containerStyle,
    buttonStyle,
    textStyle,
    rippleColor,
    underlayColor,
    children,
    onPress,
  } = props
  return (
    <View style={[style.container, containerStyle]}>
      {Platform.OS === "android" ? (
        <TouchableNativeFeedback
          onPress={onPress}
          background={TouchableNativeFeedback.Ripple(
            rippleColor || colorStyle.semiLight,
            true,
          )}
        >
          <View style={[style.button, buttonStyle]} {...props}>
            <Text style={[fontStyle.button, textStyle]}>{children}</Text>
          </View>
        </TouchableNativeFeedback>
      ) : (
        <TouchableHighlight onPress={onPress} underlayColor={underlayColor}>
          <View style={[style.button, containerStyle]} {...props}>
            <Text style={[fontStyle.button, textStyle]}>{children}</Text>
          </View>
        </TouchableHighlight>
      )}
    </View>
  )
}

Button.propTypes = {
  containerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  buttonStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  textStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  rippleColor: PropTypes.string,
  underlayColor: PropTypes.string,
  children: PropTypes.string,
  onPress: PropTypes.func,
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    borderWidth: 0,
    borderRadius: 6,
    justifyContent: "center",
    alignItems: "center",
  },
  button: {
    flex: 1,
    borderRadius: 6,
    alignSelf: "stretch",
    alignItems: "center",
    justifyContent: "center",
    borderColor: colorStyle.semiLight,
    paddingVertical: 8,
    paddingHorizontal: 3,
  },
})

export default Button
