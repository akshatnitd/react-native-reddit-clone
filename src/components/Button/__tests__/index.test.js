import React from "react"
import renderer from "react-test-renderer"

import Button from ".."

jest.mock("Platform", () => {
  const Platform = require.requireActual("Platform")
  Platform.OS = "android"
  return Platform
})

jest.mock("TouchableNativeFeedback", () => {
  const RealComponent = require.requireActual("TouchableNativeFeedback")
  RealComponent.Ripple = () => null
  return RealComponent
})

it("renders correctly", () => {
  const tree = renderer.create(<Button>Button</Button>)
  expect(tree.toJSON()).toMatchSnapshot()
})
