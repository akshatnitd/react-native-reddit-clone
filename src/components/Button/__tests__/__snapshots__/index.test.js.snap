// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 6,
        "borderWidth": 0,
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;
